### PR TITLE
Chore/update device settings

### PR DESCRIPTION
**Pull request by tora-kozic**
### Description of Change ###

Adds the `IncydrDeviceSettings` class to manage device settings on Incydr devices, ie. computers without backup settings.  Previously, trying to get the device settings of an Incydr machine threw an error because we assume backup configuration values would be provided.

Created a whole new class instead of trying to reorganize the existing backup-related code to accommodate it.

- upgraded black to v22.3.0 which caused a lot of small changes
- upgraded CLA-assistant to latest version which will take affect on any future PRs 

### Testing Procedure ###

Users should be able to use the `DeviceService` methods `get_settings()` and `update_settings()` to view properties and update the _notes_ and _external_reference_ fields on the device settings of Incydr devices.

### PR Checklist ###
Did you remember to do the below?

- [x] Add unit tests to verify this change
- [x] Add an entry to CHANGELOG.md describing this change
- [x] Add docstrings for any new public parameters / methods / classes

### CODE REVIEW

**Comment by tora-kozic** [comment]
Upgrading black made a lot of small changes to files - relevant changes are in these
- docs/userguides/devicesettings.md
- src/py42/clients/settings/device_settings.py
- src/py42/services/devices.py
- tests/clients/settings/test_incydr_device_settings.py
- tests/services/test_devices.py

File: src/py42/clients/settings/device_settings.py [hunk 1]
```diff
@@ -18,6 +18,60 @@
 )
 
 
+class IncydrDeviceSettings(UserDict):
+    """Class used to manage individual Incydr devices.  These devices have no backup settings and are only the **notes** and **external reference** fields are modifiable."""
```
Comment (timabrmsn) [inline src/py42/clients/settings/device_settings.py:22]: and ~~are~~ only

File: src/py42/services/devices.py [hunk 2]
```diff
@@ -267,20 +268,25 @@ def get_settings(self, guid):
             :class:`py42.clients.settings.device_settings.DeviceSettings`: A class to help manage device settings.
         """
         settings = self.get_by_guid(guid, incSettings=True)
-        return DeviceSettings(settings.data)
+        print(settings.data)
```
Comment (timabrmsn) [inline src/py42/services/devices.py:271]: whoops

File: tests/clients/settings/test_incydr_device_settings.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    )
+    def test_device_settings_properties_return_expected_value_and_cannot_be_changed(
+        self, param
+    ):
+        name, expected_value = param
+        assert getattr(self.device_settings, name) == expected_value
+        with pytest.raises(AttributeError):
+            setattr(self.device_settings, name, expected_value)
+
+    @pytest.mark.parametrize("param", [("notes", None), ("external_reference", None)])
```
Comment (timabrmsn) [inline tests/clients/settings/test_incydr_device_settings.py:76]: We should add examples with real test strings here too, as we could hypothetically change the implementation to something like:

```
@property
    def notes(self):
        pass
```
and this test would still pass.

File: tests/services/test_devices.py [hunk 4]
```diff
@@ -163,3 +166,67 @@ def test_get_page_when_org_not_found_raises_expected_error(
...
+        settings = client.get_settings("42")
+        assert isinstance(settings, DeviceSettings)
+
+    def test_get_settings_returns_incydr_settings_when_artemis_service(
+        self, mocker, mock_connection
+    ):
+        text = """{"service": "Artemis"}"""
+        # response = """{'computerId': 3985, 'name': '994627866755117009', 'osHostname': 'DESKTOP-I2SKEML', 'guid': '994627866755117009', 'type': 'COMPUTER', 'status': 'Active', 'active': True, 'blocked': False, 'alertState': 0, 'alertStates': ['OK'], 'userId': 6340, 'userUid': '994627866352463825', 'orgId': 3332, 'orgUid': '985187827481212202', 'computerExtRef': None, 'notes': None, 'parentComputerId': None, 'parentComputerGuid': None, 'lastConnected': '2021-02-17T20:00:06.959Z', 'osName': 'win', 'osVersion': 'DontKnow', 'osArch': 'DontKnow', 'address': None, 'remoteAddress': None, 'javaVersion': '', 'modelInfo': '', 'timeZone': None, 'version': 0, 'productVersion': '1.1.1970', 'buildVersion': None, 'creationDate': '2021-02-17T20:00:06.994Z', 'modificationDate': '2021-02-17T20:00:06.994Z', 'loginDate': '2021-02-17T20:00:06.959Z', 'service': None, 'orgSettings': {'securityKeyLocked': False}}"""
```
Comment (timabrmsn) [inline tests/services/test_devices.py:185]: probably don't need this line anymore?

**Review by timabrmsn — CHANGES_REQUESTED** [review]

**Review by timabrmsn — APPROVED** [review]

**Pull request merged**